core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -239,9 +240,9 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
   }
 
   private def validateStatusRequestArguments(): Unit = {
-    if (!master.startsWith("spark://")) {
+    if (!master.startsWith("spark://") && !master.startsWith("mesos://")) {
```

Comment:
technically not needed since we already do another level of validation in the client, but OK to keep